src/oph/ehoks/palaute/opiskelija.clj

```diff
@@ -87,23 +87,6 @@
         (palaute/already-initiated? existing-heratteet)
         [nil herate-basis :jo-lahetetty-talla-rahoituskaudella]
 
-        (and (= kysely :aloituskysely)
-             (added? :osaamisen-hankkimisen-tarve prev-hoks hoks))
```

Comment:
🤔 Sinänsä aloin vielä miettimään tätä `osaamisen-hankkimisen-tarve`:tta, eli jos se on ollut alun perin false ja muutetaan myöhemmin trueksi niin silloin palautteen pitäisi lähteä muodostumaan (jos muut ehdot täyttyvät). Eikös nyt tuo `palaute/already-initiated?` estä palautteen muodostumisen tässä tapauksessa?

Reply:
Tuossa ylempänä on tuo tarkistus
```
   (not (:osaamisen-hankkimisen-tarve hoks))
   [:ei-laheteta :osaamisen-hankkimisen-tarve :ei-ole]
```
eli tuossa sun esimerkissä, jos `osaamisen-hankkimisen-tarve` on alunperin ollut `false`, niin tuolloin hoksin ensimmäisellä tallennuskerralla ei pitäisi tallentua herätettä. Kun myöhemmin päivitetään `osaamisen-hankkimisen-tarve` arvoon `true`, niin tuon yllä olevan tarkistuksen ei pitäisi enää estää palautteen muodostumista, eikä `palaute/already-initiated?` pitäisi myöskään estää, kun olemassa olevaa herätettä ei ole.

Reply:
Ok, kiitos selvennyksestä! Olin siinä käsityksessä että `:ei-laheteta` -tilaisetkin talletuvat palautteet-tauluun ja jos sellainen on ennestään niin sitten palautteen alustamista ei yritetä uudelleen.

Reply:
Joo, näin mäkin luulin aluksi kun tutkailin noita tilasiirtymiä: https://wiki.eduuni.fi/download/attachments/415248744/palaute-tilat.png?version=3&modificationDate=1699880764950&api=v2
Tuota kaaviota voisi täsmentää tuolta osin, eli `ei_laheteta` ei olisi enää terminaalitila, ja siitä lähtisi nuoli takaisin `odottaa_kasittelya`. Noihin tilasiirtymiä kuvaaviin nuoliin voisi myös kirjoittaa selitteet, mitkä kaikki tekijät voi triggeröidä minkäkin tilasiirtymän. Silloin jo pelästään tuo kaavio dokumentoisi paljon palauteprosessin toimintaa.

Ehdottaisin myös, että kaavioon lisättäisiin eilisessä palaverissa puhuttu `ei_yhteystietoja` tjms. Tulisi varmaankin seuraavanaliset tilasiirymät

* `kysely_muodostettu -> ei_yhteystietoja`, silloin kun oltaisiin tekemässä viestien lähetystä, mutta kyselyyn vastaavan molemmat yhteystiedot puuttuvat
* `ei_yhteystietoja -> kysely_muodostettu`, silloin kun HOKSiin päivitetään sähköposti tai puhelinnumero (tai molemmat) ja kyselyssä on vastausaikaa jäljellä.
* `ei_yhteystietoja -> vastausaika_loppunut` ja `ei_yhteystietoja -> vastattu`, varmaan jonkun ajastetun palaute-backend ajon yhteydessä voitaisiin tehdä näihin liittyvät tarkistukset ja em. tilasiirtymät. Periaatteessahan palautteen voisi vaan jättää tilaan `ei_yhteystietoja`, mutta toisaalta silloin ei voida saada kovinkaan totuudenmukaisia lukemia kyselyistä, joihin on vastattu tai joihin on jätetty kokonaan vastaamatta.